### ISSUE TITLE
[VL] Enable ccache in Gluten CPP build

### ISSUE THREAD

**Issue by PHILO-HE** [apache/incubator-gluten#7873]
### Description

It's for speeding up subsequent compilation. Currently, it is not time-consuming to build Gluten CPP. Just created this issue for possibly enabling at some time point.

**Comment by FelixYBW** [comment]
Gluten CPP is still OK. From our current script, velox will be fetched and recompile each time, which is time consuming. We may make the script more smart, like skip download if Velox is already cloned, does incremental velox build automatically, instead of replying on velox's commit ID.

**Comment by PHILO-HE** [comment]
> Gluten CPP is still OK. From our current script, velox will be fetched and recompile each time, which is time consuming. We may make the script more smart, like skip download if Velox is already cloned, does incremental velox build automatically, instead of replying on velox's commit ID.

@FelixYBW, yes, currently it doesn't require much time to build Gluten CPP code. Will continuously optimize Velox build.
